web/src/pages/PreviewPage/PreviewPage.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useRef, useState, useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { useReactToPrint } from 'react-to-print';
+import styles from './PreviewPage.module.css';
+import Header from '../../components/Header/Header.jsx';
+import CVPreview from '../../components/CVPreview/CVPreview.jsx';
+
+const PreviewPage = () => {
+  const { state } = useLocation();
+  console.log(state)
+  const navigate = useNavigate();
+  const printRef = useRef();
+
+  const [isPrinting, setIsPrinting] = useState(false);
+  const { cvData, personalInfo } = state || {};
+
+  const [currentCvData, setCurrentCvData] = useState(() => cvData || null);
+  const [currentPersonalInfo, setCurrentPersonalInfo] = useState(() => personalInfo || {});
+  console.log(typeof currentCvData)
+  console.log(personalInfo)
+  const promiseResolveRef = useRef(null);
+
+
+  if (!currentCvData) {
+  return (
+    <div style={{ padding: '2rem', color: 'red', textAlign: 'center' }}>
+      <h2>Something went wrong.</h2>
+      <p>We couldn’t load your enhanced CV. Please try again.</p>
+    </div>
+  );
+}
+  const transformedCvData = {
+  fullName: currentPersonalInfo?.fullName || '',
+  contact: {
+    email: currentPersonalInfo?.email || '',
+    phone: currentPersonalInfo?.phone || '',
+    linkedin: currentPersonalInfo?.linkedin || '',
+    github: currentPersonalInfo?.github || '',
+    portfolio: currentPersonalInfo?.portfolio || ''
+  },
+  professional_summary: currentCvData?.professionalSummary || '',
```

Comment:
this will cause issues as the `CVPreview` component is expecting a variable called `professionalSummary` I recommend changing `professional_summary` to `professionalSummary` here